### PR TITLE
[CARBONDATA-535]Support data type: date and char

### PR DESCRIPTION
**Pull request by QiangCai**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/68/

File: examples/spark2/src/main/resources/data.csv [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-1,11,100,44.4,flink,2015/5/23,23.23
-1,10,150,43.4,spark,2015/7/24,254.12
-1,10,100,47.4,spark,2015/7/23,876.14
-3,14,160,43.4,hive,2015/7/26,3454.32
-2,10,100,43.4,impala,2015/7/23,456.98
-1,10,100,43.4,spark,2015/5/23,32.53
-4,16,130,42.4,impala,2015/7/23,67.23
-1,10,100,43.4,spark,2015/7/23,832.23
\ No newline at end of file
+1,10,100,48.4,spark,2015/4/23,1.23,2015/4/23,aaa
```
Comment (jackylk) [inline examples/spark2/src/main/resources/data.csv]: can you add hour, minute and second for timestampField?

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/directdictionary/TimestampDataTypeDirectDictionaryTestCase.scala [hunk 2]
```diff
@@ -69,86 +76,134 @@ class TimestampDataTypeDirectDictionaryTest extends QueryTest with BeforeAndAfte
       val csvFilePath = currentDirectory + "/src/test/resources/datasample.csv"
       sql("LOAD DATA local inpath '" + csvFilePath + "' INTO TABLE directDictionaryTable OPTIONS" +
         "('DELIMITER'= ',', 'QUOTECHAR'= '\"')");
+//      sql("select * from directDictionaryTable").show
```
Comment (jackylk) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/directdictionary/TimestampDataTypeDirectDictionaryTestCase.scala]: remove commented code

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/directdictionary/TimestampDataTypeDirectDictionaryTestCase.scala [hunk 3]
```diff
@@ -69,86 +76,134 @@ class TimestampDataTypeDirectDictionaryTest extends QueryTest with BeforeAndAfte
...
       sql("LOAD DATA local inpath '" + csvFilePath + "' INTO TABLE directDictionaryTable_hive");
+
     } catch {
       case x: Throwable => CarbonProperties.getInstance()
         .addProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT, "dd-MM-yyyy")
     }
   }
 
   test("test direct dictionary for not null condition") {
+//    checkAnswer(
```
Comment (jackylk) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/directdictionary/TimestampDataTypeDirectDictionaryTestCase.scala]: remove commented code

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/123/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/125/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/180/

**Comment by QiangCai** [comment]
@ravipesala 
Please correct CI to support -Pbuild-with-format.

**Pull request closed without merging**